trailofbits/osquery
# Fix m1 integration tests

This PR fixes M1 integration tests, osquery/osquery#7360.

| Table | Fix | Notes |
|------|----|-------|
| `python_packages` |  Add missing Python installation directory | The result set was empty for the M1 test system and I verified that the current path set being checked was still accurate for M1. I added the XCode command line tools Python package to the search directory list, which resolved the issue. |
| `kernel_info` | Change the test to allow `device` column, `boot-device-path` property, optional | This property appears to be missing on M1 systems. |
| `nvram` | Change the test to allow values be empty strings | The `boot-args` property is an empty string in both nvram and the kernel_info tables, which I verified as accurate with the `ioreg` output |
| `preferences` | Relax the test query to allow more domain values | The existing test query was returning empty results if no custom system preference was set for the current user or if the test was not run as root. The new test query still executes quickly and should return results for more systems. |

### Validation Tests

- Run the integration tests on Mac Intel and verify that all pass successfully.
- Run the integration tests on Mac M1 and verify that all pass successfully.

```
ctest -R tests_integration_tables-test -V
```
